de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ArchiveIncomingProjectNegotiation.java

```diff
@@ -101,16 +117,34 @@ protected void transfer(IProgressMonitor monitor,
 
         // the host do not send an archive if we do not need any files
         if (filesMissing) {
-            receiveAndUnpackArchive(projectMapping, transferListener, monitor);
+            receiveAndUnpackArchive(projectMapping, archiveTransferListener,
+                monitor);
         }
     }
 
+    @Override
+    protected void cleanup(IProgressMonitor monitor,
+        Map<String, IProject> projectMapping) {
+        /*
+         * TODO Move disable queuing responsibility to SarosSession (see todo
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=160&issues=AWTLb7qM6dA5b-W27MG0&open=AWTLb7qM6dA5b-W27MG0)

[ISSUE_KEY]: AWTLb7qM6dA5b-W27MG0